File: backend/app/models/message.py
from datetime import datetime
from bson import ObjectId

class Message:
    def __init__(self, conversation_id: ObjectId, sender: str, content: str, 
                 audio_path: str = None, transcription: str = None, feedback_id: str = None):
        self._id = ObjectId()
        self.conversation_id = conversation_id
        self.sender = sender  # "user" or "ai"
        self.content = content
        self.audio_path = audio_path
        self.transcription = transcription
        self.feedback_id = feedback_id
        self.timestamp = datetime.utcnow()

    def to_dict(self):
        return {
            "_id": self._id,
            "conversation_id": self.conversation_id,
            "sender": self.sender,
            "content": self.content,
            "audio_path": self.audio_path,
            "transcription": self.transcription,
            "feedback_id": self.feedback_id,
            "timestamp": self.timestamp
        }

File: mobile_clean_architecture/lib/features/conversations/domain/entities/message.dart
import 'package:equatable/equatable.dart';

/// Enum representing who sent a message
enum SenderType {
  /// Message sent by the user
  user,
  
  /// Message sent by the AI
  ai,
}

/// Entity representing a message in a conversation
class Message extends Equatable {
  /// Unique identifier for the message
  final String id;
  
  /// ID of the conversation this message belongs to
  final String conversationId;
  
  /// Who sent this message
  final SenderType sender;
  
  /// The content of the message
  final String content;
  
  /// When the message was sent
  final DateTime timestamp;
  
  /// Path to the audio recording (if available)
  final String? audioPath;
  
  /// Transcription of the audio recording (if available)
  final String? transcription;
  
  /// ID of the associated feedback (if available)
  final String? feedbackId;

  const Message({
    required this.id,
    required this.conversationId,
    required this.sender,
    required this.content,
    required this.timestamp,
    this.audioPath,
    this.transcription,
    this.feedbackId,
  });

  @override
  List<Object?> get props => [
    id, 
    conversationId, 
    sender, 
    content, 
    timestamp, 
    audioPath, 
    transcription, 
    feedbackId,
  ];
}
